x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -332,7 +336,18 @@ export function XYChart({
         }
         legendPosition={legend.position}
         showLegendExtra={false}
-        theme={chartTheme}
+        theme={{
+          ...chartTheme,
+          ...(shouldEnrichThemeWithValueLabels && {
+            barSeriesStyle: {
+              displayValue: {
+                fill: 'white',
+                offsetX: 0,
+                offsetY: 0,
```

Comment:
As long as we're positioning the text labels inside the bars, I think offsetX and offsetY should be set to negative values depending on the chart type. The text is hard to read when it's touching the edge.

Reply:
I see where this is coming from and I understand the aim of it: the negative value on the `offset` may have some issues in some particular scenarios (like with very small/tiny bars), but it makes sense to me.
I proposed also an alternative approach on the `elastic-charts` repository which makes value labels positioning a bit more sensible, and even reproduce the same Visualize behaviour (horizontal and vertical centering on the bar): https://github.com/elastic/elastic-charts/pull/784

I've discussed a bit with @markov00 about it and perhaps there's some margin to add even more options there.

The current `offset` code here is more a placeholder rather than final code.